armor-write/src/main/java/com/rapid7/armor/store/S3WriteStore.java

```diff
@@ -61,6 +64,12 @@
   private final ShardStrategy shardStrategy;
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
   private static final String INTERVAL_TAG = "interval";
+  private static final String TENANT_CACHE_KEY = "tenants";
```

Comment:
I don't understand why you need a key just for tenants..I think you can simplfy this just by caching tenants that have been stored on s3. This avoid an s3 check.